emscripten.py

```diff
@@ -1691,27 +1691,9 @@ def create_receiving(function_table_data, function_tables_defs, exported_impleme
     runtime_assertions = ''
   else:
     runtime_assertions = RUNTIME_ASSERTIONS
-    # assert on the runtime being in a valid state when calling into compiled code. The only exceptions are
-    # some support code
-    receiving_functions = [f for f in exported_implemented_functions if f not in ('_memcpy', '_memset', '_emscripten_replace_memory', '__start_module')]
-
-    wrappers = []
-    for name in receiving_functions:
-      wrappers.append('''\
-var real_%(name)s = asm["%(name)s"];
-asm["%(name)s"] = function() {%(runtime_assertions)s
-  return real_%(name)s.apply(null, arguments);
-};
-''' % {'name': name, 'runtime_assertions': runtime_assertions})
-    receiving = '\n'.join(wrappers)
 
   shared.Settings.MODULE_EXPORTS = module_exports = exported_implemented_functions + function_tables(function_table_data)
 
-  # Currently USE PTHREADS + EXPORT ES6 doesn't work with the previously generated assertions.
-  # ES6 modules disallow re-assigning read-only properties.
-  if shared.Settings.USE_PTHREADS and shared.Settings.EXPORT_ES6 and shared.Settings.ASSERTIONS and receiving:
-    receiving = "assert(!ENVIRONMENT_IS_PTHREAD, 'Error: USE_PTHREADS and EXPORT_ES6 requires ASSERTIONS=0');\n" + receiving
```

Comment:
Is this is an unrelated change?

Reply:
Sort of. I added this assertion originally because I couldn't get `USE_PTHREADS` + `EXPORT_ES6` working with `ASSERTIONS`. This PR fixes the underlying issue, so this isn't needed any more.